extensions/common/aws/aws-s3-core/src/main/java/org/eclipse/edc/aws/s3/S3ClientRequest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       ZF Friedrichshafen AG - initial implementation
+ *
+ */
+
+package org.eclipse.edc.aws.s3;
+
+import org.eclipse.edc.connector.transfer.spi.types.SecretToken;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+
+public class S3ClientRequest {
```

Comment:
can't this be a `record`? Which will save from having getters, to string, equal and hash code